guac_actix/src/network_endpoints.rs

```diff
@@ -0,0 +1,112 @@
+use actix_web::http::Method;
+use actix_web::*;
+
+use guac_core::channel_client::types::{Channel, UpdateTx};
+
+use guac_core::crypto::CryptoService;
+use guac_core::{CRYPTO, STORAGE};
+
+use guac_core::channel_client::ChannelManager;
+
+use failure::Error;
+
+use futures::Future;
+
+use NetworkRequest;
+
+use guac_core::counterparty::Counterparty;
+use std::net::SocketAddr;
+
+pub fn init_server(port: u16) {
+    server::new(|| {
+        App::new()
+            .resource("/update", |r| r.method(Method::POST).with(update_endpoint))
+            .resource("/propose", |r| {
+                r.method(Method::POST).with(propose_channel_endpoint)
+            })
+            .resource("/channel_created", |r| {
+                r.method(Method::POST).with(channel_created_endpoint)
+            })
+            .resource("/channel_joined", |r| {
+                r.method(Method::POST).with(channel_joined_endpoint)
+            })
+    }).bind(&format!("[::0]:{}", port))
+        .unwrap()
+        .start();
+}
+
+pub fn update_endpoint(
+    update: Json<NetworkRequest<UpdateTx>>,
+) -> Box<Future<Item = Json<UpdateTx>, Error = Error>> {
+    trace!("got state update {:?}", update);
+    Box::new(
+        STORAGE
+            .get_channel(update.from_addr.clone())
+            .and_then(move |mut channel_manager| {
+                channel_manager.received_payment(&update.data)?;
+                Ok(Json(channel_manager.create_payment()?))
+            })
+            .responder(),
+    )
+}
+
+pub fn propose_channel_endpoint(
+    (req, channel): (HttpRequest, Json<NetworkRequest<Channel>>),
```

Comment:
I don't think I've ever seen this syntax, so this is just a de-tupling expression done in the param list?

Reply:
yeah